### PR TITLE
Fix for unpickling polar plot issue #4068

### PR DESCRIPTION
**Pull request by olanmatt**
This implements a fix for the bug documented in issue #4068 

`PolarAffine` of `lib/matplotlib/projections/polar.py` returned an empty state when being pickled because of an unimplemented `__getstate__` function. Removing this function resolved the reported exception and restored the expected behaviour.

### CODE REVIEW

**Comment by jenshnielsen** [comment]
Travis failure in docs build is not related to this Pull request. Fixed by #4267

**Comment by stallone1993** [comment]
@jenshnielsen Thanks for the heads up! Would it be possible to get the tests restarted?

**Comment by tacaswell** [comment]
cc @pelson 

It looks like Phil added this code in 7885d944 in PR https://github.com/matplotlib/matplotlib/pull/1175

**Comment by stallone1993** [comment]
cc @pelson @tacaswell 

Here's what we've noticed.

The problem arises because `PolarAffine` is overriding the `__getstate__()` function of `Transform`, but all it does is return an empty dictionary currently. As such, when Pickle is used to restore the instance of a `PolarAffine`, all of its attributes 
are lost and not set. Therefore, the `AttributeError` exception was being raised.

It seems like other children of `Transform` seem to just use their parent's `__getstate__` and `__setstate__` functions - which seem to work well with pickle.

**Comment by tacaswell** [comment]
Can you add a test?  https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/tests/test_pickle.py has some tests already (and while you are in there can you add an `@cleanup` to all of the tests?).

File: lib/matplotlib/projections/polar.py [hunk 1]
```diff
@@ -112,10 +112,6 @@ def get_matrix(self):
         return self._mtx
     get_matrix.__doc__ = Affine2DBase.get_matrix.__doc__
 
-    def __getstate__(self):
-        return {}
-
-
```
Comment (pelson) [inline lib/matplotlib/projections/polar.py]: Nit-pick: We leave 2 newlines between top level module objects. (I think the PEP8 tests will probably catch this)

**Comment by pelson** [comment]
I'm OK with the change. We can't be certain that this wont break other classes, which may have attributes which aren't picklable, but we can deal with that as the occur (should be a simple fix when it crops up).

:+1:

**Comment by dkua** [comment]
Whitespace issue fixed, @cleanup decorator added to tests, and a minimum test created.

There was a funny error in one of the Travis test suites, does it simply need a restart?

**Comment by tacaswell** [comment]
Yes, I restarted that test.  Looks like apt timed out or some-such.

**Comment by tacaswell** [comment]
Thank you!

**Comment by tacaswell** [comment]
cherry-picked to color_overhaul as 592818a175edf46aa497bec9b42a4333dd449467

**Pull request merged**